spec/lib/authorization-builder.spec.js

```diff
@@ -0,0 +1,102 @@
+        loggerMock = require("../mock/logger-mock")();
+        dateMock = jasmine.createSpyObj("date", ["now"]);
+        dateMock.now.andReturn("NOW");
+        container = require("../../lib/container")("api.opentoken.io");
+        container.register("dateService", dateMock);
+        container.register("logger", loggerMock);
+        AuthorizationBuilder = container.resolve("AuthorizationBuilder");
+        StreamReadable = container.resolve("StreamReadable");
+    });
+    describe(".createHeader", () => {
+        var body, code, expectedAuth, requestOptions, secret, signature;
+
+
+        /**
+         * @param {string} content
+         * @return {stream~Readable}
+         */
+        function createStream(content) {
+            var stream;
+
+            stream = new StreamReadable();
+            stream.push(content);
+            stream.push(null);
+
+            return stream;
+        }
+        beforeEach(() => {
+            code = "code";
+            secret = "secret";
+            body = "hello world";
+            signature = "49c0edb8558bee14649dc77da6ab934a23477261549dd377f19206f85a56785b";
+            requestOptions = {
+                headers: {
+                    "content-type": "text/plain",
+                    "x-opentoken-date": "NOW",
+                    host: "api.opentoken.io"
+                },
+                method: "POST",
+                url: "https://api.opentoken.io/account/abc123/token?public"
+            };
+            expectedAuth = `OT1-HMAC-SHA256-HEX; access-code=code; signed-headers=content-type x-opentoken-date host; signature=${signature}`;
+        });
+        it("will use a body that is a string", () => {
+            var builder;
+
+            builder = new AuthorizationBuilder(code, secret, requestOptions, body);
+
+            return builder.createHeader().then((auth) => {
+                expect(auth).toEqual(expectedAuth);
+            });
+        });
+        it("will use a body that is a readable stream", () => {
+            var builder;
+
+            body = createStream(body);
+            builder = new AuthorizationBuilder(code, secret, requestOptions, body);
+
+            return builder.createHeader().then((auth) => {
+                expect(auth).toEqual(expectedAuth);
+            });
+        });
+        it("will standardize header keys", () => {
+            var builder;
+
+            requestOptions.headers = {
+                "CoNtent-tYpe": "text/plain",
+                "x-opEntOken-datE": "NOW",
+
+                // host is a special case and is lowercased specifically.
+                host: "Api.OPENtoken.io"
+            };
+
+            builder = new AuthorizationBuilder(code, secret, requestOptions, body);
+
+            // Important to note: The fact that the "signed-headers" are not
+            // cased the same as when they were used to build the signature
+            // is OK. So long as the ones were lowercase when used to build
```

Comment:
You should use the right pronoun.  "ones" is for individuals or the number.  Not as a reference for a group.  The right one would be "they" as you used in the previous line.